h/models/document.py

```diff
@@ -412,6 +412,8 @@ def merge_documents(session, documents, updated=None):
     duplicates = documents[1:]
     duplicate_ids = [doc.id for doc in duplicates]
 
+    log.info(f"Merging {len(duplicate_ids) + 1} documents")
```

Comment:
You could also do `documents.count()` but it causes all kinds of problems with some really hairy tests for this file.

Reply:
There's a general default of not composing the string with logging, so that if it doesn't kick in it doesn't get evaluated like:

```
    log.info(f"Merging %s documents", len(duplicate_ids) + 1)
```
First things first, _this doesn't matter_, I'm just trying to work out why this is.

I can't really see this saving any time for a couple of reasons:

 * This is getting evaluated anyway (maybe if it was a lambda or something it might not?)
 * Old style string interpolation is likely slower than just a fixed string with a new one anyway

Reply:
Done. I think this is because if the log level is set to critical or warning etc in the logging config, then `log.info()`'s don't do anything. So you can save time by not formatting those strings. In practice it won't make any difference but :man_shrugging: